src/main/java/com/coveo/pushapiclient/PlatformClient.java

```diff
@@ -82,17 +82,21 @@ public PlatformClient(String apiKey, String organizationId, Environment environm
      * Create a new push source
      *
      * @param name             The name of the source to create
+     * @param sourceType
+     * @param isPushEnabled
+     * @param isStreamEnabled
      * @param sourceVisibility The security option that should be applied to the content of the source. See [Content Security](https://docs.coveo.com/en/1779).
      * @return
      * @throws IOException
      * @throws InterruptedException
      */
-    public HttpResponse<String> createSource(String name, SourceVisibility sourceVisibility) throws IOException, InterruptedException {
+    public HttpResponse<String> createSource(String name, final String sourceType, final boolean isPushEnabled, final boolean isStreamEnabled, SourceVisibility sourceVisibility) throws IOException, InterruptedException {
```

Comment:
yep a source can have both attributes , yep we can deduce them from type , but we will loose some flexibility since  they are different parameters in the api  , so a push souce will be just push enabled and catalog source is push and stream enabled ?

Reply:
Can a catalog source be only StreamEnable? Are there any specials licenses for commerce that prevent one property or the other to be true?